File: elemental/src/dto/iterazionedto.ts
import {UserDTO} from "./userdto";
import {PostDTO} from "./postdto";

export class IterazioneDTO {
  id: number

  likes: number

  dataCommento: Date

  contenuto: string

  imgURL: string

  user: UserDTO

  post: PostDTO

  constructor(
  id: number,

  likes: number,

  dataCommento: Date,

  contenuto: string,

  imgURL: string,

  user: UserDTO,

  post: PostDTO,
  ) {
    this.id = id
    this.likes = likes
    this.dataCommento = dataCommento
    this.contenuto = contenuto
    this.imgURL = imgURL
    this.user = user
    this.post = post
  }
}

File: src/main/java/com/simonecampis/Elemental/dto/IterazioneDTO.java
package com.simonecampis.Elemental.dto;

import com.simonecampis.Elemental.model.Post;
import com.simonecampis.Elemental.model.User;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Date;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class IterazioneDTO {

    private Long id;

    private Long likes;

    private Date dataCommento;

    private String contenuto;

    private String imgURL;

    private User user;

    private Post post;

}
